core/src/main/java/io/grpc/Metadata.java

```diff
@@ -147,6 +147,10 @@ public Metadata(byte[]... binaryValues) {
     checkArgument(binaryValues.length % 2 == 0,
         "Odd number of key-value pairs: %s", binaryValues.length);
     for (int i = 0; i < binaryValues.length; i += 2) {
+      // The transport might provide an array with null values at the end.
+      if (binaryValues[i] == null) {
```

Comment:
FYI: I was planning on using the Internal Accessor stuff that I used in method descriptor to short cut into creating these objects.  

I am going to be looking into changing this class to do something like that in the near future.